actiontext/app/javascript/actiontext/attachment_upload.js

```diff
@@ -20,13 +20,14 @@ export class AttachmentUpload {
 
   directUploadDidComplete(error, attributes) {
     if (error) {
-      throw new Error(`Direct upload failed: ${error}`)
+      let uploadFailed = new CustomEvent("trix-attachment-error", {"detail": {"error": error, "attachment": this.attachment }})
+      this.element.dispatchEvent(uploadFailed)
```

Comment:
Let's find a way to either push this upstream and have Active Storage dispatch the event for us. Or, find a way to pull in Active Storage's event utilities so we maintain the same signature and don't have to construct the entire `detail` payload manually. https://github.com/rails/rails/blob/3d0a7822a5dc8828a32c9bf032daa950223364a2/activestorage/app/javascript/activestorage/direct_upload_controller.js#L44-L48 https://github.com/rails/rails/blob/3d0a7822a5dc8828a32c9bf032daa950223364a2/activestorage/app/javascript/activestorage/helpers.js#L25-L41